### DISCUSSION TITLE
Is there any way to have the auto-cropping disabled after stitching?

### DISCUSSION THREAD

**Discussion by Faris-Faiz** [OpenStitching/stitching#239] [Q&A] [answered]
I looked up online, the term is called stitching errors. Here's an example of stitching errors:
![image](https://github.com/user-attachments/assets/335bead6-4fbb-4981-8e8a-cbb9b1f19462)

The image from stitching using this library:
![image](https://github.com/user-attachments/assets/bf8f70a4-93b5-4033-896b-68cf784b2dbd)

I'm currently doing research on using generative AI inpainting to fill in those black parts using conditional GANs. Is it possible to have the stitching errors be maintained after stitching using this library?

**Comment by lukasalexanderweber** [comment]
Just use --no-crop

**Reply by Faris-Faiz** [reply]
> Just use --no-crop

Is there a way to do this programmatically, along the lines of passing as a parameter in the settings? 

Or do you recommend I use Python's subprocess to invoke this command: `stitch img_dir/IMG*.jpg --no-crop` through the CLI

**Reply by lukasalexanderweber** [reply, marked as answer]
Stitcher(crop=false)

**Reply by Faris-Faiz** [reply]
> Stitcher(crop=false)

Thank you for your kind response, I actually found the documentation based on the [Python Notebook here](https://github.com/OpenStitching/stitching_tutorial/blob/master/Stitching%20Tutorial.ipynb). Thank you very  much for your attention and I apologize if the question seems repetitive

**Reply by lukasalexanderweber** [reply]
No worries, good luck with your project!

**Reply by lukasalexanderweber** [reply]
If you find a useful approach feel free to share it e.g. in https://github.com/OpenStitching/stitching/discussions/categories/show-and-tell

**Reply by Faris-Faiz** [reply]
With Stitcher(crop=False):
![image](https://github.com/user-attachments/assets/9d1c45f3-f434-4270-b018-5392b15952bd)


With Stitcher(crop=True):
![image](https://github.com/user-attachments/assets/d7c8f063-07a5-4d98-9cba-fabf52022a2f)

I notice that there is some significant difference in the quality of stitching/seam blending when cropping is turned off, is that intentional?

> If you find a useful approach feel free to share it e.g. in https://github.com/OpenStitching/stitching/discussions/categories/show-and-tell

Will do!

**Reply by lukasalexanderweber** [reply]
In our reseach we noticed that cropping **before** calculating seam lines significantly improves the seam lines. This is best explained with this comparison:

![09_result_with_seam_lines](https://github.com/user-attachments/assets/5f3d31f6-c6cc-4d97-9e00-87a6aeb0c926)
![09_result_with_seam_lines](https://github.com/user-attachments/assets/88284add-46b6-4799-99e3-7b6610140e7d)

when cropped, the best seam can be found anywhere. without cropping, the seams are forced to be between the transitions of the images
